File: languages/go/entity/identity/multichoice_attribute_descriptor.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type MultichoiceAttributeDescriptor struct {
	AttributeDescriptor
	AllowedValues []string
}

func NewMultichoiceAttributeDescriptor(name string, technicalName string, allowedValues []string, description string) MultichoiceAttributeDescriptor {
	a := AttributeDescriptor{
		DisplayName:   name,
		TechnicalName: technicalName,
		Description:   description,
	}
	return MultichoiceAttributeDescriptor{
		AttributeDescriptor: a,
		AllowedValues:       allowedValues,
	}
}

func NewMultichoiceAttributeDescriptorFromProto(s *proto.MultiChoiceAttributeDefinition) MultichoiceAttributeDescriptor {
	if s == nil {
		return MultichoiceAttributeDescriptor{}
	}
	a := AttributeDescriptor{
		DisplayName:   s.GetId(),
		TechnicalName: s.Id,
		Description:   s.Description,
	}
	return MultichoiceAttributeDescriptor{
		AttributeDescriptor: a,
		AllowedValues:       s.AllowedValues,
	}
}

func (s MultichoiceAttributeDescriptor) ToProto() *proto.MultiChoiceAttributeDefinition {
	return &proto.MultiChoiceAttributeDefinition{
		DisplayName:   s.DisplayName,
		Id:            s.TechnicalName,
		Description:   s.Description,
		AllowedValues: s.AllowedValues,
	}
}

File: languages/js/src/entity/identity/multichoice_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities";
import { AttributeDescriptor } from "./attribute_descriptor";

export class MultichoiceAttributeDescriptor extends AttributeDescriptor {
  values: string[];

  constructor(
    displayName: string,
    technicalName: string,
    values: string[],
    description?: string
  ) {
    super(displayName, technicalName, description);
    this.values = values;
  }

  public toProto(): identityEntitiesProto.MultiChoiceAttributeDefinition {
    return identityEntitiesProto.MultiChoiceAttributeDefinition.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      allowedValues: this.values
    });
  }

  static fromProto(
    r: identityEntitiesProto.MultiChoiceAttributeDefinition
  ): MultichoiceAttributeDescriptor {
    return new MultichoiceAttributeDescriptor(
      r.displayName,
      r.id,
      r.allowedValues,
      r.description
    );
  }
}
